turbinia/notify.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+'''Sends email notifications'''
+
+import logging
+import smtplib
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText
+from turbinia import config
+log = logging.getLogger('turbinia')
+
+#Sends notfications via email
+def sendmail(subject, message):
+  '''Sends an Email notification via SMTP'''
+  try:
+    if config.EMAIL_NOTIFICATIONS is True:
+
+      #Puts message in MIME format
+      msg = MIMEMultipart()
+      msg['From'] = config.EMAIL_ADDRESS
+      msg['To'] = config.EMAIL_RECIEVING_ADDRESS
+      msg['Subject'] = subject
+      msg.attach(MIMEText(message))
+
+      #start SMTP
+      server = smtplib.SMTP(config.EMAIL_HOST_ADDRESS, config.EMAIL_PORT)
+
+      # identify to mail server
+      server.ehlo()
+
+      # secure email using TLS
+      server.starttls()
+
+      # reidentify as using TLS
+      server.ehlo()
+
+      #Attempt to login
+      try:
+        server.login(config.EMAIL_ADDRESS, config.EMAIL_PASSWORD)
+      except NameError:
+        log.info(
+            'EMAIL_ADDRESS or EMAIL_PASSWORD is not definied, ' +
+            'attempting to continue without logging in')
+
+      server.sendmail(
+          config.EMAIL_ADDRESS, config.EMAIL_RECIEVING_ADDRESS, msg.as_string())
+      log.info('Email notification sent')
+    else:
+      log.info('Email notifications are disabled')
+
+  except smtplib.SMTPException as e:
+    log.error(e)
+    log.error(
+        'Email failed to send, SMTP has raised an error,' +
+        'this likely means that their is a problem with the config')
+  except TypeError:
+    log.error('Email failed to send, There is likely a problem with the config')
```

Comment:
Can you log the exception string here (and for the exception below) as well?  That way it's easier to tell what the exact problem is.  A common thing to do is to put this at the end of the line like:
```suggestion
    log.error('Email failed to send, there is likely a problem with the config: {0!s}'.format(exception))
```

Another minor nit, but maybe lower-case the word `There` here (or use a period before it to start a new sentence).   Same for `A value` in the error message below.  Thanks!